plugins/parsers/avro/testdata/union-array/expected.out

```diff
@@ -0,0 +1 @@
+array,name=pi data_0=3,data_1=3.0999999046325684,data_2=3.140000104904175,data_3=3.1410000324249268 1682509200092000
```

Comment:
is the float parsing happening in the upstream library? Is there a reason for the loss?

Reply:
Well, the reason for the loss is that we're parsing it as a 32-bit float (because that's what the schema says), but rendering it into the metric with 15 digits of claimed precision because the InfluxDB Float is inherently 64-bit...so the bottom 32 bits are all zero, and that is not exactly the same as the nice decimal value (which also isn't exactly the same, floating point numbers aren't real numbers, etc., etc.).

Go Playground gives the same results:

```
package main

import (
	"fmt"
	"strconv"
)

func main() {
	vals := []string{"3", "3.1", "3.14", "3.141"}
	for _, v := range vals {
		f, err := strconv.ParseFloat(v, 32)
		if err != nil {
			fmt.Printf("Error: %v\n", err)
		}
		fmt.Printf("string %s -> float %.15f\n", v, f)
	}
}
```

I can make the test suite cleaner by making the input values "double" if you want, which would give us the correct answer to 15 digits, but it is in fact the case that if you're sending 32-bit floats (which are a legal Avro data type), you're going to have some imprecision once it's in line protocol format.

This is, I think, correct (but surprising) float-conversion behavior.

Reply:
> This is, I think, correct (but surprising) float-conversion behavior.

Yep, you are right, I missed the 32-bit.